elements/lisk-chain/test/unit/event.spec.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+/* eslint-disable no-bitwise */
+
+import { codec } from '@liskhq/lisk-codec';
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { eventSchema } from '../../src/schema';
+import { Event } from '../../src/event';
+import { EVENT_TOPIC_HASH_LENGTH_BYTES, EVENT_TOTAL_INDEX_LENGTH_BYTES } from '../../src/constants';
+
+describe('event', () => {
+	const eventObj = {
+		moduleID: Buffer.from([0, 0, 0, 2]),
+		typeID: Buffer.from([0, 0, 0, 1]),
+		topics: [getRandomBytes(32), getRandomBytes(20), getRandomBytes(2)],
```

Comment:
Don't we have specific length for a topic? If so, maybe we should implement a byte length check?

Reply:
maximum length of a topic is 4, but each topic can be any length